libraries/exporters/src/main/java/com/siemens/sw360/exporter/ComponentExporter.java

```diff
@@ -55,21 +94,62 @@ public int getColumns() {
         }
 
         @Override
-        public List<String> makeRow(Component component) {
-            List<String> row = new ArrayList<>(getColumns());
+        public SubTable makeRows(Component component) {
+            return extendedByReleases
+                    ? makeRowsWithReleases(component)
+                    : makeRowForComponentOnly(component);
+        }
 
-            row.add(nullToEmpty(component.name));
-            row.add(joinStrings(component.languages));
-            row.add(joinStrings(component.categories));
-            row.add(joinStrings(component.operatingSystems));
-            row.add(joinStrings(component.softwarePlatforms));
-            row.add(nullToEmpty(component.createdBy));
-            row.add(nullToEmpty(component.createdOn));
-            row.add(joinStrings(getVersions(component.releases)));
+        protected SubTable makeRowsWithReleases(Component component) {
+            releases = getReleases(component);
+            SubTable table = new SubTable();
 
+            if(releases.size() > 0) {
+                for (Release release : releases) {
+                    List<String> currentRow = makeRowForComponent(component);
```

Comment:
you could move `makeRowForComponent` out, since it is called by both branches. It might then be good to make it immutable or treat it like such a thing.